modules/devshell.nix

```diff
@@ -98,7 +98,7 @@ let
 
     # If the file is sourced, skip all of the rest and just source the env
     # script.
-    if [[ $0 != "''${BASH_SOURCE[0]}" ]]; then
+    if (return 0) &>/dev/null; then
```

Comment:
Nice, I didn't know about this trick

Reply:
I can't remember where I first came across this, but I've found it to be the most reliable way to answer "am I being sourced?" under Bash.